shell/src/main/java/org/apache/kafka/shell/glob/GlobVisitor.java

```diff
@@ -93,32 +96,33 @@ public String toString() {
     }
 
     @Override
-    public void accept(MetadataNodeManager.Data data) {
+    public void accept(MetadataShellState state) {
         String fullGlob = glob.startsWith("/") ? glob :
-            data.workingDirectory() + "/" + glob;
+            state.workingDirectory() + "/" + glob;
         List<String> globComponents =
             CommandUtils.stripDotPathComponents(CommandUtils.splitPath(fullGlob));
-        if (!accept(globComponents, 0, data.root(), new String[0])) {
+        if (!accept(globComponents, 0, state.root(), new String[0])) {
             handler.accept(Optional.empty());
         }
     }
 
-    private boolean accept(List<String> globComponents,
-                           int componentIndex,
-                           MetadataNode node,
-                           String[] path) {
+    private boolean accept(
+        List<String> globComponents,
+        int componentIndex,
+        MetadataNode node,
+        String[] path
+    ) {
```

Comment:
Similar to above, do we need a null check on `node`?

Reply:
Like above, I'll add a check and helpful exception.